cfg/std.cfg

```diff
@@ -5636,6 +5641,7 @@ The obsolete function 'gets' is called. With 'gets' you'll get a buffer overrun
   </function>
   <!-- int toupper(int c); -->
   <function name="toupper,std::toupper">
+    <pure/>
```

Comment:
This is not `pure` as it depends on the locale which might change during run-time.

Reply:
Yet we use `pure` as our (only) indicator if a function has side effects.
You can also change the floating-point rounding mode, which affects the results of certain math functions...

Reply:
> You can also change the floating-point rounding mode, which affects the results of certain math functions...

True. If you would limit this assumption to only the current scope it seems like you could apply that information but as it depends on a C library function which might be called in another library function that might lead to unexpected effects. I wonder if the compiler will consider this or if this is considered like something as a "data race" or if you are calling `chdir()` somewhere in the code meaning that all bets are off.